catalog/ui/catalog-ui-filter/src/main/java/org/codice/ddf/catalog/ui/filter/builder/tools/PropertyValueNodeSupplier.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.filter.builder.tools;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * The {@link PropertyValueNodeSupplier} collects two, and only two, filter nodes and converts them
+ * to a single node. This is useful for most terminal types.
+ *
+ * @param <T> the type being used to represent a node in the filter tree.
+ */
+public class PropertyValueNodeSupplier<T> implements NodeSupplier<T> {
+  private final NodeReducer<T> reducer;
+  private T propertyNode = null;
+  private T valueNode = null;
+
+  public PropertyValueNodeSupplier(final NodeReducer<T> reducer) {
+    notNull(reducer);
+    this.reducer = reducer;
+  }
+
+  @Nullable
+  @Override
+  public NodeSupplier<T> getParent() {
+    // Always return null since most terminal nodes, unlike functions, do not require nesting
```

Comment:
These nodes are "lazy builder" objects, and we create a sort of "linked list" structure from these lazy builders to support the contract on `FlatFilterBuilder`. So for this snippet: 

```
<Filter>
  <PropertyIsEqualTo>
    <ValueReference>elevation</ValueReference>
    <Function name="stdev">
      <Literal>5</Literal>
      <Function name="lookup">
        <Literal>key</Literal>
      </Function>
      <Literal>0.03</Literal>
    </Function>
  </PropertyIsEqualTo>
</Filter>
```

The very first entity to be built is the "lookup" function, followed by the "stdev" function. While they are being built, everything else is still technically "in-progress" because we can't finish a node until all its children are finished. 

Individual implementations of `NodeSupplier` can enforce their own rules to match that of the XML schema so we know if we've called a `FlatFilterBuilder` method out of turn. In the case of the `PropertyValueNodeSupplier` we would only use those when functions are not present, so we have no need to chain them. Thus the parent can always be null for this implementation. 

See javadoc on `NodeSupplier` for more information.